lib/oppen.rb

```diff
@@ -12,13 +12,16 @@ module Oppen
   # Entry point of the pretty printer.
   #
   # @param config [Config]
+  # @param genspace [Callable] space generator: can be a lambda, a block, ...
   # @param margin [Integer] maximum line width desired
   # @param new_line [String] the delimiter between lines
+  # @param out [Buffer] should have a write method
   # @param tokens [Array[Token]] the list of tokens to be printed
   #
   # @return [StringIO] output of the pretty printer
-  def self.print(config: Config.oppen, margin: 80, new_line: "\n", tokens: [])
-    printer = Printer.new margin, new_line, config
+  def self.print(config: Config.oppen, genspace: ->(n) { ' ' * n },
+                 margin: 80, new_line: "\n", out: StringIO.new, tokens: [])
+    printer = Printer.new margin, new_line, config, genspace, out
```

Comment:
We want the genspace, but what we want from an interfact pov is to either accept a string or a lambda generator.